.evergreen/auth_oidc/azure/test.py

```diff
@@ -8,11 +8,13 @@
 _AUTH_MAP["MONGODB-OIDC"] = _authenticate_oidc
 
 app_id = os.environ['AZUREOIDC_CLIENTID']
+client_id = os.environ['AZUREOIDC_TOKENCLIENT']
 
 def callback(client_info, server_info):
```

Comment:
Optional: The side effect of setting `client_id` on `callback` is fairly confusing. Is there any way to encode the new `client_id` requirement in the `callback` function, or in a `callback` creator function?

E.g.
```python
def callback_for(app_id, client_id):
    def callback(client_info, server_info):
        ...

    return callback
```